[1.19] kots 1.19.0 release notes (#258)
replicatedhq/kots.io: content/release-notes/1.19.0.md

---
date: 2020-09-14
linktitle: "1.19.0"
title: "1.19.0"
kubernetes: "1.16, 1.17, and 1.18"
---

{{<features>}}
  * Experimental support has been added to [use an OCI Registry instead of S3](/kotsadm/installing/stateful-components) for KOTS Object Store.
{{</features>}}

{{<changes>}}
  * Helm render errors will now be included in the skippedFiles directory in the View Files tab.
{{</changes>}}

{{<fixes>}}
  * Empty files will no longer be included in the skippedFiles directory in the View Files tab.
  * Fixed support of `helm.sh/hook-delete-policy` annotation. Fixed transpiling to `kots.io/hook-delete-policy`.
  * Fixed a bug that caused Admin Console pods to fail with `ImagePullBackOff` errors after an [upgrade in airgapped installs](/kotsadm/updating/updating-admin-console/#airgap-installations).
  * Fixed an issue where some errors were not logged when restoring a snapshot.
  * Fixed a bug that caused [`kots upstream upgrade`](/kots-cli/upstream/) command to fail with error 500.

>   * FIxed a bug that caused license installation to fail if app had missing or malformed [Application spec](/reference/v1beta1/application/).

Reviewer: ```suggestion
  * Fixed a bug that caused license installation to fail if app had missing or malformed [Application spec](/reference/v1beta1/application/).
```

---
date: 2020-09-14
linktitle: "1.19.0"
title: "1.19.0"
kubernetes: "1.16, 1.17, and 1.18"
---

{{<features>}}

>   * Support for airgapped [headless installs](/kotsadm/installing/automating/).

Reviewer 2: Is there a PR to update that page for airgapped? Or perhaps should it be included in this PR so it's merged at the same time?

Reviewer 3: I think this is it: https://github.com/replicatedhq/kots.io/pull/271

---
date: 2020-09-14
linktitle: "1.19.0"
title: "1.19.0"
kubernetes: "1.16, 1.17, and 1.18"
---

{{<features>}}

>   * Support for airgapped [headless installs](/kotsadm/installing/automating/).

Reviewer 2: ```suggestion
  * Support for airgapped [headless installs](/kotsadm/installing/automating/) to existing cluster. Note, upgrading is not support headless yet.
```

>   * Support resumable uploads for airgap bundles in Kotsadm UI.

Reviewer 2: ```suggestion
  * More robust handle of airgap bundle uploads in the Admin Console, including error handling and ability to resume upload.
```

>   * Added support for additional authentications methods including OIDC and IBM Cloud to `kubectl kots`.

Reviewer 2: ```suggestion
  * Added support for `kubectl kots` to use additional authentications methods including OIDC and IBM Cloud.
```

>   * Added support for troubleshoot.sh/v1beta2 SupportBundle specs.

Reviewer 4: do we need a link here?

Reviewer 2: Also, these are two somewhat separate things. Version, and kind: SupportBundle.

Reviewer 2: ```suggestion
  * Support for `avpiVersion: troubleshoot.sh/v1beta2`
  * Support for `kind: SupportBundle`.
```

{{</features>}}

{{<changes>}}

>   * The KOTS Admin Console no longer includes the Node API pod. The functionality has been migrated into the existing Go API pod, and the resulting release bundle size is ~33% smaller and contains 1 fewer pod. All upgrades will remove this pod.

Reviewer 3: ```suggestion
  * The KOTS Admin Console no longer includes the Node API deployment. The functionality has been migrated into the existing Go API deployment, and the resulting release bundle size is 200MB smaller (from 663MB to 463MB) and contains 1 fewer deployment. All upgrades will remove this deployment.
```

  * Helm render errors will now be included in the skippedFiles directory in the View Files tab.

>   * KOTS will no longer require a namespace when configuring the registry settings, allowing for anonymous registries.

Reviewer 2: ```suggestion
  * KOTS will no longer require a namespace when configuring the registry settings.
```

  * Preflight checks using the [textAnalyze](https://troubleshoot.sh/docs/analyze/regex/) function can now run against multiple files when the provided filepath is a [glob](https://golang.org/pkg/path/filepath/#Glob).
  * Preflight checks relying on the [storageClass](https://troubleshoot.sh/docs/analyze/storage-class/) function can check for the presence of a default storageclass by leaving `storageClassName` empty.

> {{</changes>}}

Reviewer 2: Missing some more troubleshoot things. e.g. Copy directory.

Reviewer 2: imagepullsecrets...

Reviewer 2: * https://github.com/replicatedhq/troubleshoot/pull/253
* https://github.com/replicatedhq/troubleshoot/pull/249
* https://github.com/replicatedhq/troubleshoot/pull/248

  * Fixed a bug that caused Admin Console pods to fail with `ImagePullBackOff` errors after an [upgrade in airgapped installs](/kotsadm/updating/updating-admin-console/#airgap-installations).
  * Fixed an issue where some errors were not logged when restoring a snapshot.
  * Fixed a bug that caused [`kots upstream upgrade`](/kots-cli/upstream/) command to fail with error 500.
  * Fixed a bug that caused license installation to fail if app had missing or malformed [Application spec](/reference/v1beta1/application/).
  * Fixed a bug that caused KOTS to re-deploy the first release of the application when re-running preflight checks for that release.
  * Empty files are no longer included in yamlErrors.
  * Fixed a bug that caused the user to be redirected to the wrong version after editing the registry settings and clicking the "View Files" tab.
  * Fixed a bug that caused the `kots upload` command to fail with an empty KOTS config.yaml.
  * Fixed a bug that caused the registry settings configuration to fail when entering ECR registry credentials.
  * Fixed a bug that could cause `panic: d.nx != 0` error when images were being pushed to private registry during airgapped installs.

>   * Fixed an unclear error message when failing to parse an image ref while checking whether images were private. This error message will now include the filename that the offendin image was found within.

Reviewer 2: ```suggestion
  * Fixed an unclear error message when failing to parse an image ref while checking whether images were private. This error message will now include the filename that the offending image was found within.
```

  * Updated the generated `kustomization.yaml` when using a private registry to avoid a problem when two different tags of the same image are used.

>   * Fixed a bug where previously deployed versions in Kotsadm "Version Vistory" tab show a status of "Deployed" instead of "Previously Deployed".

Reviewer 3: ```suggestion
  * Fixed a bug where previously deployed versions in KOTS Admin Console "Version Vistory" tab show a status of "Deployed" instead of "Previously Deployed".
```

> {{</fixes>}}

Reviewer 2: * https://github.com/replicatedhq/troubleshoot/pull/246
* https://github.com/replicatedhq/troubleshoot/pull/252

---

> date: 2020-09-14

Reviewer 2: ```suggestion
date: 2020-09-11
```

  * Support for [`apiVersion: troubleshoot.sh/v1beta2`](https://troubleshoot.sh/docs/v1beta2/).
  * Support for [`kind: SupportBundle`](https://troubleshoot.sh/docs/support-bundle/introduction/).
{{</features>}}

{{<changes>}}
  * The KOTS Admin Console no longer includes the Node API deployment. The functionality has been migrated into the existing Go API deployment, and the resulting release bundle size is 200MB smaller (from 663MB to 463MB) and contains 1 fewer deployment. All upgrades will remove this deployment.
  * Helm render errors will now be included in the skippedFiles directory in the View Files tab.
  * KOTS will no longer require a namespace when configuring the registry settings.
  * Preflight checks using the [textAnalyze](https://troubleshoot.sh/docs/analyze/regex/) function can now run against multiple files when the provided filepath is a [glob](https://golang.org/pkg/path/filepath/#Glob).
  * Preflight checks relying on the [storageClass](https://troubleshoot.sh/docs/analyze/storage-class/) function can check for the presence of a default storageclass by leaving `storageClassName` empty.

>   * Support `ImagePullSecret` in troubleshoot run collector.

Reviewer 2: ```suggestion
  * Support [`imagePullSecret ` in troubleshoot run collector](https://troubleshoot.sh/docs/collect/run/#using-opaque-secrets).
```

>   * Add support for filtering by label for troubleshoot `NodeResources` analyzer.

Reviewer 2: ```suggestion
  * Add support for [filtering by label for troubleshoot `NodeResources` analyzer](https://troubleshoot.sh/docs/analyze/node-resources/#filter-by-labels).
```

>   * Include k8s events in troubleshoot `ClusterResources` collector.

Reviewer 2: ```suggestion
  * Include [Kubernetes events in troubleshoot `ClusterResources` collector](https://troubleshoot.sh/docs/collect/cluster-resources/#cluster-resourceseventsnamespacejson).
```
